tmf-transform.exs

```diff
@@ -0,0 +1,225 @@
+
+    ref_model_date =
+      title_row
+      |> Enum.find(&is_struct(&1, Date))
+      |> to_string()
+
+    title = title |> String.trim()
+
+    %{
+      :_generation_timestamp => DateTime.utc_now() |> DateTime.to_string(),
+      title: title,
+      version: ref_model_version,
+      version_date: ref_model_date
+    }
+  end
+
+  defp parse_headers(headers_a, headers_b) do
+    # There are 2 header rows, we need to combine them to handle
+    # sub sections and merged cells
+
+    {_, headers} =
+      Enum.zip_reduce(
+        headers_a,
+        headers_b,
+        {"", []},
+        fn a, b, {section, acc} ->
+          {s, h} = combine_headers(a, b, section)
+
+          # add header to accumulator list
+          {s, [acc | [h |> String.replace("\r\n", " ") |> String.trim()]]}
+        end
+      )
+
+    headers
+    |> List.flatten()
+  end
+
+  def combine_headers(a, b, section)
+  def combine_headers(a, "", _), do: {"", a}
+  def combine_headers("", b, ""), do: {"", b}
+  def combine_headers("", b, section), do: {"", concat_section_header(section, b)}
+  def combine_headers(a, b, ""), do: {a, concat_section_header(a, b)}
+
+  defp concat_section_header(section, header), do: Enum.join([section, header], " : ")
+
+  defp transform_artifacts(%{metadata: metadata, headers: headers, rows: rows}) do
+    artifacts =
+      rows
+      |> Enum.reject(fn [zone | _] -> zone == "" end)
+      |> Enum.map(fn r ->
+        # Combine the List of headers with the list of data from this row
+        {_, artifact} =
+          Enum.zip(headers, r)
+          # Now convert the list of tuples into a map
+         |> Enum.map_reduce(%{}, fn {k, v}, acc ->
+            # if the value has `\r\n` in it, split into a list of strings
+            value =
+              v
+              |> special_conversion_hack()
+              |> split_string_with_multiple_lines()
+
+            # Search for Section headers (from the headers) to create sub maps
+            k
+            |> has_section_marker?()
+            |> case do
+              true ->
+                # Need to split key into section/header
+                {true, add_section_to_artifact(acc, k, value)}
+              false ->
+                {true, acc |> Map.put(k, value)}
+            end
+          end)
+
+        artifact
+      end)
+
+    %{metadata: metadata, artifacts: artifacts}
+  end
+
+  defp has_section_marker?(str), do: String.contains?(str, " : ")
+
+  defp split_string_with_multiple_lines(str) when is_binary(str) do
+    str
+    |> String.split("\r\n")
+    |> case do
+      [any] -> any
+      any -> any
+    end
+  end
+
+  defp split_string_with_multiple_lines(other), do: other
+
+  # Handle unexpected formatting in a given XLSX version of the reference model
+  defp special_conversion_hack("2.2000000000000002"), do: "2.2"
+  defp special_conversion_hack("2.0099999999999998"), do: "2.01"
+  defp special_conversion_hack("10.050000000000001"), do: "10.05"
```

Comment:
those values are what are _actually_ in the OOXML on 18ish different occurrences. Custom number formats are set up so that these numbers are displayed in Excel as the shorter versions that these hack functions return. 

This was the quickest path to getting past it. Perhaps there is a more elegant solution. I'll add more detail to the comment explaining what is going on and why.